Add developer utilities to aid in dumping/loading of production server data into devcontainers

lunkwill42 opened the pull request:
## Scope and purpose

The old docker-compose-based development environment includes half a toolset (and no documentation on how to use it) for initializing the development database with a dump from another NAV host.

This adds the complete suite of tools + documentation to accomplish this in the devcontainer environment.

## Contributor Checklist

Every pull request should have this checklist filled out, no matter how small it is.
More information about contributing to NAV can be found in the
[Hacker's guide to NAV](https://nav.readthedocs.io/en/latest/hacking/hacking.html#hacker-s-guide-to-nav).

<!-- Add an "X" inside the brackets to confirm -->
<!-- If not checking one or more of the boxes, please explain why below each or remove the line if not applicable. -->

* [x] Added a changelog fragment for [towncrier](https://nav.readthedocs.io/en/latest/hacking/hacking.html#adding-a-changelog-entry)
* ~~[ ] Added/amended tests for new/changed code~~
* [x] Added/changed documentation
* [x] Linted/formatted the code with black and ruff, easiest by using [pre-commit](https://nav.readthedocs.io/en/latest/hacking/hacking.html#pre-commit-hooks-and-black)
* [x] The first line of the commit message continues the sentence "If applied, this commit will ...", starts with a capital letter, does not end with punctuation and is 50 characters or less long. See https://cbea.ms/git-commit/
* [x] This pull request is based on the correct upstream branch: For a patch/bugfix affecting the latest stable version, it should be based on that version's branch (`<major>.<minor>.x`). For a new feature or other additions, it should be based on `master`.
* ~~[ ] If applicable: Created new issues if this PR does not fix the issue completely/there is further work to be done~~
* ~~[ ] If this results in changes in the UI: Added screenshots of the before and after~~
* ~~[ ] If this adds a new Python source code file: Added the [boilerplate header](https://nav.readthedocs.io/en/latest/hacking/hacking.html#python-boilerplate-headers) to that file~~

<!-- Make this a draft PR if the content is subject to change, cannot be merged or if it is for initial feedback -->

hmpf commented on doc/hacking/using-devcontainers.rst:
Dumping/loading data from a remote production server is useful also outside of dev-containers so why is this piece of text here? It would be more sensible to have this as a more standalone doc and link to it from `using-devcontainers.rst`, ditto if there is a doc for using docker compose, ditto if there are docs for how to do maintenance/upgrades/backups etc. etc.

johannaengland reviewed (commented):
I agree with @hmpf. Otherwise it looks good and works as expected

johannaengland commented:
> I do sort of agree, @hmpf and @johannaengland - but where to put it? I mean, we also have https://nav.readthedocs.io/en/latest/howto/migrate-data.html - but this is slightly more geared towards end users and not so much developers (although it benefits them too)...

Either there or adding a new page to https://nav.readthedocs.io/en/latest/hacking/index.html

lunkwill42 commented:
So, I've moved the documentation of the new tools/scripts to the generic data migration guide, which already contains more extensive and/or low-level docs on how to migrate data between NAV installs.  I've updated both the devcontainer hacking howto and the docker-compose hacking-howto to reference this document instead.

Acceptable, @johannaengland and @hmpf ? :)
